pkg/build/controller/policy/serial_latest_only.go

```diff
@@ -0,0 +1,92 @@
+package policy
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/golang/glog"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+)
+
+// SerialLatestOnlyPolicy implements the RunPolicy interface. This variant of
+// the serial build policy makes sure that builds are executed in same order as
+// they were created, but when a new build is created, the previous, queued
+// build is cancelled, always making the latest created build run as next. This
+// will produce consistent results, but might not suit the CI/CD flow where user
+// expect that every commit is built.
+type SerialLatestOnlyPolicy struct {
+	BuildUpdater buildclient.BuildUpdater
+	BuildLister  buildclient.BuildLister
+}
+
+// IsRunnable implements the RunPolicy interface.
+func (s *SerialLatestOnlyPolicy) IsRunnable(build *buildapi.Build) (bool, error) {
+	bcName := buildutil.ConfigNameForBuild(build)
+	if len(bcName) == 0 {
+		return false, NewNoBuildConfigLabelError(build)
+	}
+	if err := kerrors.NewAggregate(s.cancelPreviousBuilds(build)); err != nil {
```

Comment:
I will update the godoc. Yes, it will cancel all builds and it will not look at the policy those builds have. It can also cancel parallel builds if there are some in the "new" state for some reason.